azurerm/config.go

```diff
@@ -198,10 +198,14 @@ func withRequestLogging() autorest.SendDecorator {
 }
 
 func setUserAgent(client *autorest.Client) {
-	version := terraform.VersionString()
-	client.UserAgent = fmt.Sprintf("HashiCorp-Terraform-v%s", version)
+	// if the user agent already has a value append Terraform user agent string
```

Comment:
This function seems to be a bit harder to read now, suggestions:

1. Put assignment statement in the if-clause, will it be harder for debugging coz one line of code contains too much statements ?
2. No matter tfversion or azureagent, you're appending it into user-agent, can we merge the code to avoid some kind of duplications ? Consider: what if we need to append even more fields here?
3. "HashiCorp-Terraform-v%s" seems to be a template/format string, shall we put it in constants ?
4. I can understand the second if-clause that you use assignment then checking coz they're related. But for the first if-clause, the assigment of tfVersion has nothing to do with the later checking, shall we move it to a separated line ?

Reply:
1. I did it that way because of comments from my previous PR where it was suggested to put the assignment of the variable in the if block.

2. I don't really see where it can get duplications, unless this code is called more than once, and if that happens I would consider that a bug. If you wanted to make sure there will not be duplications in the user agent string you should then also do a sub-string check before appending the text to the user-agent.

3. Yes it is a template string but it is only ever used once in the code, doesn't seem to be a good reason to make this a constant.

4. Yes, it can be it's own line, but as I stated above I did it this way because of my previous PR comments.

Reply:
#1/4, if a = xxx; a != "" is very good pattern coz it really saves one line and these 2 lines are contextual. While if b = xxx; a != "", i feel it's very weird. Who gave the suggestion ? tom ? We probably can sync with him...

#2, the duplication means the logic to append the string with ; as delimiter. If you're asked to append 10 more strings in the user-agent, you will not like to copy and paste. Anyway, it's just a suggestion.

#3 read-only and scope are different concepts, even its scope is function level, you still can use constants.